ui/components/src/BarChart/BarChart.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo } from 'react';
+import { UnitOptions, formatValue } from '@perses-dev/core';
+import { use, EChartsCoreOption } from 'echarts/core';
+import { BarChart as EChartsBarChart } from 'echarts/charts';
+import { GridComponent, DatasetComponent, TitleComponent, TooltipComponent } from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useTheme } from '@mui/material';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import { EChart } from '../EChart';
+
+use([EChartsBarChart, GridComponent, DatasetComponent, TitleComponent, TooltipComponent, CanvasRenderer]);
+
+export interface BarChartData {
+  label: string;
+  value: number | null | undefined;
+}
+
+export interface BarChartProps {
+  width: number;
+  height: number;
+  data: BarChartData[] | null | undefined;
```

Comment:
not sure if we need to do it now, but i think we need to account for multiple series in the future

Reply:
I’ve been out so I need to get caught up on all the use cases we are trying to cover because if this was supposed to cover BarGauge too, I was expecting it to be a stacked bar so thresholds could be supported. Do we have a plan for adding thresholds later, is that a separate component and then there will be logic in the panel that shows either component depending if thresholds are populated?

Reply:
Really I think BarGauge could be covered by adding panel options (visual.display: ‘radial’ as default and visual.display: ‘bar’) to GaugeChart and adding logic to *that* panel but mostly trying to understand the schema for BarChart since sounded like Karan wanted it to cover both use cases

Reply:
> not sure if we need to do it now, but i think we need to account for multiple series in the future


When you say multiple series, do you mean sub grouping? Example below
![image](https://github.com/perses/perses/assets/6935733/3d87de8d-0a9f-4a96-9af9-66b16805bc25)

I think we could extend (not needed now) this to look like this:

```ts 

type BarChartData = {
   label: string,
   grouping: [{
     label: string,
     value: number
   }]
}
```

Where "Variety" is the top level group and "Genotype" is the sub group.

Reply:
Oh yeah that's what i meant! Since this isn't time series data, does that also mean we need to update the DataQueryProvider?

Reply:
> Oh yeah that's what i meant! Since this isn't time series data, does that also mean we need to update the DataQueryProvider?

Not necessarily, you can transform a TimeSeries data interface by using 2 labels on each series. 1 label would be the top level group, and one would be the sub group. Example:


```ts
// first compute the last calculation of a timeseries
[
   { labels: { app: "app-1", device: "ios" }, lastValue: 10 },
   { labels: { app: "app-1", device: "android" }, lastValue: 20 },
   { labels: { app: "app-2", device: "ios" }, lastValue: 2 },
   { labels: { app: "app-2", device: "android" }, lastValue: 4 },
]
```

Then group them for the bar chart
```tsx
[
   { label: "app-1", grouping: [{ label: "ios", value: 10 }, { label: "android", value: 20 }] },
   { label: "app-2", grouping: [{ label: "ios", value: 2 }, { label: "android", value: 4 }] },
]
```

Reply:
ohhhh got it thanks!

Reply:
Multi series and thresholds aren't part of the MVP but I think we will likely add them later. We don't have designs for either yet so I'm not sure exactly what it would like